source/puppet/5.5/lang_data_abstract.md

```diff
@@ -117,6 +117,72 @@ Note that you can use capture groups in the regular expressions, but they won't
 * `Pattern[/\A[a-z].*/]` --- matches any string that begins with a lowercase letter.
 * `Pattern[/\A[a-z].*/, /\Anone\Z/]` --- matches the above **or** the exact string `"none"`.
 
+### `Timespan`
+
+The `Timespan` data type matches a Timespan; [strings][], or [numbers][] that represent a time and date; or `default`, which represents an infinite point in the past or future.
+
+It takes up to two parameters, and defaults to an infinite range to the past and future.
+
+#### Parameters
+
+The full signature for `Timespan` is:
+
+    Timespan[ (<TIMESPAN START OR LENGTH>, (<TIMESPAN END>)) ]
+
+Position | Parameter | Data Type | Default Value | Description
+---------|-----------|-----------|---------------|------------
+1 | Timespan Start or Length | String, Float, Integer, or `default` | `default` (-Infinity in a span) | Length of the timespan if passed alone, or the `from` value in a range if passed with a second parameter
+2 | Timespan End | String, Float, Integer, or `default` | `default` (+Infinity), or none if only one value is passed | The `to` value in a range
+
+`Timespan` values are interpreted depending on their format.
+
+* A String in the format of `D-HH:MM:SS` represents a span of days (`D`), hours (`HH`), minutes (`MM`), and seconds (`SS`)
+* An Integer or Float represents a number of seconds
+
+A `Timespan` range with either end defined as `default` (infinity) is an _open range_, while any other span is a _closed range_. The span's range is inclusive.
+
+The `Timespan` type is not enumerable.
+
+#### Examples:
+
+* `Timespan[2]` --- a timespan of 2 seconds
+* `Timespan[77.3]` --- a timespan of 1 minute, 17 seconds, and 300 milliseconds (77.3 seconds)
+* `Timespan['1-00:00:00', '2-00:00:00']` --- a timespan between 1 and 2 days
+
+### `Timestamp`
+
+The `Timestamp` data type matches a Timestamp; [strings][] or [numbers][] that represent a time and date; or `default`, which represents an infinite point in the past or future.
```

Comment:
```suggestion
The `Timestamp` data type matches Timestamp values within a given range; specified with [strings][] or [numbers][] that represents the points in time limiting the range; or `default`, which represents an infinite point in the past or future
```

Reply:
A range of time stamps are bound to points in time; from a specific date/time to another specific date/time. A range of Timestamps is a range of durations. For example - a movie may be 2h30min long (a Timespan), but that does not say on which date and time it is showing (a Timstamp), so a range of timespans may be "movies longer than 60 minutes but shorter than 4h". 

We use `default` in the type system to describe the absolute min/max values of things - we have that already in Integer and Float ranges. It works the same here; the default upper/lower range values can also be explicitly given as literal `default`. Does that make it clearer?

Reply:
BTW: the term "time" is best avoided, since it is after all relative ;-) and it does not say if you mean a "point in time" (date/time) or a "duration of time" (length). So `Timestamp` is "When", and `Timespan` is "How Long"